modules/core/src/command_line_parser.cpp

```diff
@@ -134,22 +135,25 @@ void CommandLineParser::getByIndex(int index, bool space_delete, int type, void*
                 String v = impl->data[i].def_value;
                 if (space_delete == true) v = impl->cat_string(v);
 
-                // it is an error if we just got the default value here
-                if(v.empty())
-                    break;
-
+                // the key was neither specified nor has it a default value
+                if(v.empty()) {
```

Comment:
allowing empty strings here essentially says that string arguments always have a default value. (which is logically true, an empty string is still a string)
However I would argue that while logical, it is not the expected behavior as string arguments are usually used for paths (or other non-empty strings).
Users then would have to check that the argument is non-empty and generate a custom error message instead of relying on the parser doing the right thing.

Note that you still can use a blank string as default argument calling `args.get("foo", false)`.

The problem at hand is IMO caused by an API abuse by checking the value instead of calling `args.has()`. See my 2nd commit.